### PR TITLE
Add PropertyTableRowDiffer, refs #682

### PR DESCRIPTION
**Pull request by mwjames**
- Code responsible for computing row diffs was moved to its own component
- `SMWSql3SmwIds` now always returns `int` for id's to avoid id mismatch as seen in #682

### CODE REVIEW

**Comment by mwjames** [comment]
@JeroenDeDauw @kghbln This PR should improve the reliability of the `SQLStore` (especially in connection with #682) but to avoid a situation where a regression slipped through (well the existing integration tests couldn't catch them) we should be vigilant in the upcoming period of testing when using the master to monitor any unintended behavior. 

We can't really design an integration test that would mimic the observation made in #682 therefore those edge cases need to be filtered manually.

The following counter measures have been added to avoid possible duplicate table rows by:

- Making sure that primitive types are compared to the same type
- https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1018/files#diff-3a38a0c6ec81c2c749c281edbf906b89R335
- https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1018/files#diff-6b1b07782b2e982b031ecc568ace2afaR284
- https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1018/files#diff-6b1b07782b2e982b031ecc568ace2afaR163
- The last line of defense is https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1018/files#diff-6b1b07782b2e982b031ecc568ace2afaR117 (`$hashModifier` is currently unused, only to be used if the above changes still show that hashes are not updated according to the state of the data)

File: src/SQLStore/PropertyTableRowDiffer.php [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+ * @author Markus Krötzsch
+ * @author Nischay Nahata
+ * @author mwjames
+ */
+class PropertyTableRowDiffer {
+
+	/**
+	 * @var Store
+	 */
+	private $store = null;
```
Comment (JeroenDeDauw) [inline src/SQLStore/PropertyTableRowDiffer.php]: No need to set this to null, constructor always sets it

**Comment by JeroenDeDauw** [comment]
Did a superficial review and looks good. Thanks for all the work on this!

**Comment by mwjames** [comment]
Well, nothing is going to change from having it here so let's be bold ...
